cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,7 +60,44 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS
+                 /usr
+                 /usr/local
+                 /opt
+                 $ENV{HOME}
+                 $ENV{HOME}/.local
+                 $ENV{HOME}/.local/opt
+                 $ENV{HOME}/bin)
+
+    set(PACKAGE_FIND_VERSION ${Zephyr-sdk_FIND_VERSION})
+    set(COMPATIBLE_SDK_VERSION 0.0.0)
+
+    # Load the version file of each potential Zephyr SDK version and check if it
+    # is greater than the existing best found version.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_CONFIGS})
+      string(REGEX REPLACE "\.cmake$" "Version.cmake" ZEPHYR_SDK_VERSION_CANDIDATE ${ZEPHYR_SDK_CANDIDATE})
+
+      include(${ZEPHYR_SDK_VERSION_CANDIDATE} NO_POLICY_SCOPE)
```

Comment:
not fond of bypassing `find_package()` infrastructure this way when loading the Version file directly.

Such implementation should be used with great care.
Reason why it's used in https://github.com/zephyrproject-rtos/zephyr/blob/main/share/zephyr-package/cmake/zephyr_package_search.cmake is because that code handles multiple Zephyr within same workspace, and or Zephyrs which may not be registered in the CMake package registry.

The `FindZephyr-sdk.cmake` should only deal with Zephyr SDK in common search paths or those that are registered in the CMake package registry.

Why can't `find_package()` own infrastructure be used ?
For example by simply call `find_package(Zephyr-sdk <version-to-try>)` from highest version and down until a Zephyr SDK itself reports it is compatible.

Reply:
Fixed